libpod/events/journal_linux.go

```diff
@@ -273,3 +295,53 @@ func GetNextEntry(ctx context.Context, j *sdjournal.Journal, stream bool, untilT
 		return entry, nil
 	}
 }
+
+func compress(s string) (string, error) {
+	if len(s) < 1024 {
+		return s, nil
+	}
+
+	var buff bytes.Buffer
+
+	w, err := xz.NewWriter(&buff)
+	if err != nil {
+		return "", fmt.Errorf("cannot create compression writer: %w", err)
+	}
+
+	_, err = io.Copy(w, strings.NewReader(s))
+	if err != nil {
+		return "", fmt.Errorf("cannot copy to compression writer: %w", err)
+	}
+
+	err = w.Close()
+	if err != nil {
+		return "", fmt.Errorf("cannot close compression writer: %w", err)
+	}
+
+	return buff.String(), nil
+}
+
+func decompress(s string) (string, error) {
+	if len(s) >= 6 &&
+		s[0] == 0xFD &&
+		s[1] == 0x37 &&
+		s[2] == 0x7A &&
+		s[3] == 0x58 &&
+		s[4] == 0x5A &&
+		s[5] == 0x00 {
```

Comment:
Doesn't this squick anyone out? `ulikunitz/xz` does not provide an `IsXZ()` mechanism, tsk tsk, but I would hope that we could have some consideration for future maintainers here, and offer one ourselves, using human-readable chars instead of cryptic hex.